assets/admin/students/student-bulk-action-button/index.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+import { render, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import StudentModal from '../student-modal';
+
+/**
+ *  Student bulk action button.
+ */
+export const StudentBulkActionButton = () => {
+	const [ action, setAction ] = useState( 'add' );
+	const [ isModalOpen, setIsModalOpen ] = useState( false );
+	const [ studentIds, setStudentIds ] = useState( [] );
+	const closeModal = () => setIsModalOpen( false );
+	const setActionValue = ( selectedValue ) => {
+		switch ( selectedValue ) {
+			case 'enrol_restore_enrolment':
+				setAction( 'add' );
+				break;
+			case 'remove_enrolment':
+				setAction( 'remove' );
+				break;
+			case 'remove_progress':
+				setAction( 'reset-progress' );
+				break;
+			default:
+		}
+	};
+
+	const openModal = () => {
+		const hiddenSenseiBulkAction = document.getElementById(
+			'bulk-action-selector-top'
+		);
+
+		const hiddenSelectedUserIdsField = document.getElementById(
+			'bulk-action-user-ids'
+		);
+
+		if ( hiddenSenseiBulkAction ) {
+			setActionValue( hiddenSenseiBulkAction.value );
+		}
+
+		if ( hiddenSelectedUserIdsField ) {
+			try {
+				setStudentIds( JSON.parse( hiddenSelectedUserIdsField.value ) );
+			} catch ( e ) {}
```

Comment:
Probably this try-catch block [was introduced here](https://github.com/Automattic/sensei/pull/5004/commits/c09076cf73ac684d08d4c507761d3b0ca5568ebe) to fix the tests as the hiddenSelectedUserIdsField was not available while running tests. But if setStudentIds throws an exception and no student is selected, should we let the modal open? We can try adding a mock bulk-action-user-ids In the test and removing the try catch block. WDYT?